aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -1,26 +1,40 @@
 <%
 require 'active_support/core_ext/numeric/bytes'
 require 'cdo/aws/cloudfront'
-unless system("git ls-remote --exit-code origin #{branch} > /dev/null")
-  raise 'Current branch needs to be pushed to GitHub with the same name, otherwise deploy will fail.
-To specify an alternate branch name, run `rake stack:start branch=BRANCH`.'
+
+origin = "'https://github.com/code-dot-org/code-dot-org.git'"
+
+if rack_env?(:adhoc) && RakeUtils.git_branch == branch
+  # Current branch is the one we're deploying to the adhoc server,
+  # so check whether it's up-to-date with the remote before we get any further.
+  unless `git remote show #{origin} 2>&1 | grep \"(up to date)\" | grep \"#{branch}\" | wc -l`.strip.to_i > 0
+    raise 'Current adhoc branch needs to be up-to-date with GitHub branch of the same name, otherwise deploy will fail.
+To specify an alternate branch name, run `rake adhoc:start branch=BRANCH`.'
+  end
```

Comment:
This was brought in from the standalone template. I added the `rack_env?(:adhoc)` conditional so non-adhoc builds aren't affected.